tools/autograd/derivatives.yaml

```diff
@@ -1127,13 +1127,24 @@
   values: gather_with_keepdimed_indices(self_t, dim, indices, keepdim)
 
 - name: mul.Tensor(Tensor self, Tensor other) -> Tensor
-  self: mul_tensor_backward(grad, other, self.scalar_type())
-  other: mul_tensor_backward(grad, self, other.scalar_type())
-  result: other_t * self_p + self_t * other_p
+  dispatch:
+    Default:
+      self: mul_tensor_backward(grad, other, self.layout(), self.sizes(), self.scalar_type())
+      other: mul_tensor_backward(grad, self, other.layout(), other.sizes(), other.scalar_type())
+      result: other_t * self_p + self_t * other_p
+    AutogradNestedTensor:
+      self: mul_tensor_backward(grad, other, self.layout(), IntArrayRef(), self.scalar_type())
+      other: mul_tensor_backward(grad, self, other.layout(), IntArrayRef(), other.scalar_type())
+      result: other_t * self_p + self_t * other_p
```

Comment:
@albanD , @cpuhrsch , are you fine with that, or would you recommend something else, i.e. a separate function?